packages/mobile/src/onboarding/registration/JoinCelo.tsx

```diff
@@ -0,0 +1,252 @@
+import { ScrollView, StyleSheet } from 'react-native'
+import * as RNLocalize from 'react-native-localize'
+import SafeAreaView from 'react-native-safe-area-view'
+import { connect } from 'react-redux'
+import { setName, setPhoneNumber, setPromptForno } from 'src/account/actions'
+import { PincodeType } from 'src/account/reducer'
+import { hideAlert, showError } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import DevSkipButton from 'src/components/DevSkipButton'
+import { Namespaces, withTranslation } from 'src/i18n'
+import { nuxNavigationOptions } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { RootState } from 'src/redux/reducers'
+
+interface StateProps {
+  cachedName: string | null
+  cachedNumber: string | null
+  cachedCountryCallingCode: string | null
+  pincodeType: PincodeType
+  acceptedTerms: boolean
+}
+
+interface DispatchProps {
+  setPromptForno: typeof setPromptForno
+  showError: typeof showError
+  hideAlert: typeof hideAlert
+  setPhoneNumber: typeof setPhoneNumber
+  setName: typeof setName
+}
+
+type OwnProps = StackScreenProps<StackParamList, Screens.JoinCelo>
+
+type Props = StateProps & DispatchProps & WithTranslation & OwnProps
+
+interface State {
+  name: string
+  nationalPhoneNumber: string
+  e164Number: string
+  isValidNumber: boolean
+  countryCodeAlpha2: string
+}
+
+const mapDispatchToProps = {
+  setPromptForno,
+  setPhoneNumber,
+  setName,
+  showError,
+  hideAlert,
+}
+
+const mapStateToProps = (state: RootState): StateProps => {
+  return {
+    cachedName: state.account.name,
+    cachedNumber: state.account.e164PhoneNumber,
+    cachedCountryCallingCode: state.account.defaultCountryCode,
+    pincodeType: state.account.pincodeType,
+    acceptedTerms: state.account.acceptedTerms,
+  }
+}
+
+function getPhoneNumberState(
+  phoneNumber: string,
+  countryCallingCode: string,
+  countryCodeAlpha2: string
+) {
+  const phoneDetails = parsePhoneNumber(phoneNumber, countryCallingCode)
+
+  if (phoneDetails) {
+    return {
+      nationalPhoneNumber: phoneDetails.displayNumber,
+      e164Number: phoneDetails.e164Number,
+      isValidNumber: true,
+      countryCodeAlpha2: phoneDetails.regionCode!,
+    }
+  } else {
+    return {
+      nationalPhoneNumber: phoneNumber,
+      e164Number: '',
+      isValidNumber: false,
+      countryCodeAlpha2,
+    }
+  }
+}
+
+export class JoinCelo extends React.Component<Props, State> {
+  static navigationOptions = nuxNavigationOptions
+
+  countries = new Countries(this.props.i18n.language)
+
+  state: State = {
+    name: this.props.cachedName || '',
+    ...getPhoneNumberState(
+      this.props.cachedNumber || '',
+      this.props.cachedCountryCallingCode || '',
+      RNLocalize.getCountry()
+    ),
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps: Props) {
```

Comment:
Don't judge me 😅 
This will be eliminated by switching to a function component. 
I wanted to keep the class component here for now to prevent unnecessary changes from the existing code.

Reply:
Why not put it in componentDidUpdate? Though we waste a render, we avoid using a deprecated method

Reply:
Good point. Changing this.